src/Common/ItechArt.Common/OperationResult.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+
+namespace ItechArt.Common;
+
+public class OperationResult<TResult, TError>
+    where TError : Enum
+{
+    private readonly TError _error;
+    private readonly TResult _result;
+
+    private readonly bool _success;
```

Comment:
decide if we should manage 'success' state with flag, or should it be calculated